src/state/sqlx.rs

```diff
@@ -0,0 +1,34 @@
+use serde::Serialize;
+use sqlx::{
+    postgres::{PgArgumentBuffer, PgTypeInfo, PgValueRef},
+    types::Json,
+    Database, Decode, Encode, Postgres, Type,
+};
+
+use crate::State;
+
+impl Type<Postgres> for State {
+    fn type_info() -> PgTypeInfo {
+        <Json<State> as Type<Postgres>>::type_info()
+    }
+
+    fn compatible(ty: &<Postgres as Database>::TypeInfo) -> bool {
+        *ty == <Json<State> as Type<Postgres>>::type_info()
+    }
```

Comment:
I'm not 100% sure how it is supposed to be done, but I believe you should not compare types like this. Maybe wrapping types into an enum would work better for compile-time type checks?

Reply:
It's sqlx black magic. I'm not sure if it's the right way